File: subarraysum2.cpp
#include <iostream>
#include <unordered_map>
#include <vector>

int main()
{
    using namespace std;

    int n, x;
    cin >> n >> x;

    vector<int> arr(n);
    for (int i = 0; i < n; ++i)
    {
        cin >> arr[i];
    }

    int prefix = 0;
    unordered_map<int, int> check;
    check[0] = 1;
    int cnt = 0;

    for (int num : arr)
    {
        prefix += num;
        cnt += check[prefix - x];
        check[prefix]++;
    }

    cout << cnt << endl;

    return 0;
}

File: subarraysum2.py
import sys
from collections import defaultdict

input = sys.stdin.read

def main():
    data = input().split()
    x = int(data[1])
    arr = map(int, data[2:])

    prefix = 0
    check = defaultdict(int)
    check[0] = 1
    cnt = 0

    for num in arr:
        prefix += num
        cnt += check[prefix - x]
        check[prefix] += 1

    sys.stdout.write(str(cnt) + '\n')

if __name__ == "__main__":
    main()